nip47/controllers/create_connection_controller.go

```diff
@@ -0,0 +1,134 @@
+	ExpiresAt         *uint64                `json:"expires_at"` // unix timestamp
+	Isolated          bool                   `json:"isolated"`
+	Metadata          map[string]interface{} `json:"metadata,omitempty"`
+}
+
+type createConnectionResponse struct {
+	// pubkey is given, user requesting already knows relay.
+	WalletPubkey string `json:"wallet_pubkey"`
+}
+
+func (controller *nip47Controller) HandleCreateConnectionEvent(ctx context.Context, nip47Request *models.Request, requestEventId uint, publishResponse publishFunc) {
+	params := &createConnectionParams{}
+	resp := decodeRequest(nip47Request, params)
+	if resp != nil {
+		publishResponse(resp, nostr.Tags{})
+		return
+	}
+
+	logger.Logger.WithFields(logrus.Fields{
+		"request_event_id": requestEventId,
+		"params":           params,
+	}).Info("creating app")
+
+	var expiresAt *time.Time
+	if params.ExpiresAt != nil {
+		expiresAtUnsigned := *params.ExpiresAt
+		expiresAtValue := time.Unix(int64(expiresAtUnsigned), 0)
+		expiresAt = &expiresAtValue
+	}
+
+	maxAmountSat := params.MaxAmount / 1000
+
+	// explicitly do not allow creating an app with create_connection permission
+	if slices.Contains(params.RequestMethods, models.CREATE_CONNECTION_METHOD) {
+		publishResponse(&models.Response{
+			ResultType: nip47Request.Method,
+			Error: &models.Error{
+				Code:    constants.ERROR_INTERNAL,
+				Message: "cannot create a new app that has create_connection permission via NWC",
+			},
+		}, nostr.Tags{})
+		return
+	}
+
+	supportedMethods := controller.lnClient.GetSupportedNIP47Methods()
+	if slices.ContainsFunc(params.RequestMethods, func(method string) bool {
+		return !slices.Contains(supportedMethods, method)
+	}) {
+		publishResponse(&models.Response{
+			ResultType: nip47Request.Method,
+			Error: &models.Error{
+				Code:    constants.ERROR_INTERNAL,
+				Message: "One or more methods are not supported by the current LNClient",
+			},
+		}, nostr.Tags{})
+		return
+	}
+	scopes, err := permissions.RequestMethodsToScopes(params.RequestMethods)
+
+	supportedNotificationTypes := controller.lnClient.GetSupportedNIP47NotificationTypes()
+	if len(params.NotificationTypes) > 0 {
+		if slices.ContainsFunc(params.NotificationTypes, func(method string) bool {
+			return !slices.Contains(supportedNotificationTypes, method)
+		}) {
+			publishResponse(&models.Response{
+				ResultType: nip47Request.Method,
+				Error: &models.Error{
+					Code:    constants.ERROR_INTERNAL,
+					Message: "One or more notification types are not supported by the current LNClient",
+				},
+			}, nostr.Tags{})
+			return
+		}
+		scopes = append(scopes, constants.NOTIFICATIONS_SCOPE)
+	}
+
+	// ensure there is at least one scope
+	if len(scopes) == 0 {
+		publishResponse(&models.Response{
+			ResultType: nip47Request.Method,
+			Error: &models.Error{
+				Code:    constants.ERROR_INTERNAL,
+				Message: "No methods provided",
+			},
+		}, nostr.Tags{})
+		return
+	}
```

Comment:
Can this be moved to `CreateApp` itself?

Reply:
this is difficult - the apps service deals with scopes, not request_methods or notification_types, and by doing that we will have unclear error messages for requests via NWC. There might be a small amount I can move.

Reply:
Ah then just moving the `len(scopes) == 0` check should be good ig